File: baekjoon/js/11726.js
const fs = require("fs");
const buf = fs.readFileSync(0, "utf-8").toString().trim();

const n = Number(buf);

const count = Array(n + 1).fill(0);
count[1] = 1;
count[2] = 2;

for (let i = 3; i <= n; i++) {
  count[i] = (count[i - 1] + count[i - 2]) % 10007;
}

const result = count[n];

console.log(result);

File: baekjoon/rust/11726.rs
use std::io::stdin;

fn main() {
    let mut buf = String::new();
    stdin().read_line(&mut buf).unwrap();

    let n: usize = buf.trim().parse().unwrap();

    let mut prev2 = 1;
    let mut prev1 = 2;
    let mut result = 0;

    if n == 1 {
        result = prev2;
    } else if n == 2 {
        result = prev1;
    } else {
        for _ in 3..=n {
            result = (prev1 + prev2) % 10007;
            prev2 = prev1;
            prev1 = result;
        }
    }

    print!("{}", result);
}
